### PR TITLE
Add `consumerID` to Pulsar pub/sub

### PR DESCRIPTION
**Pull request by hhunter-ms**
## Description

Add to `consumerID` to the pulsar component example and spec table

## Issue reference

PR will close: #3211

### CODE REVIEW

**Comment by msfussell** [comment]
@berndverst - Is a consumer Id created by Dapr runtime if one is not supplied in the metadata?

**Comment by berndverst** [comment]
@msfussell I don't know anything about this component. That being said - yes according to https://github.com/dapr/components-contrib/blob/master/pubsub/pulsar/pulsar.go#L370

The `subscriptionName` which I suspect is the equivalent of a consumer group is using the `consumerID` metadata attribute -- and `consumerID` always gets injected by runtime if not directly supplied via component metadata.

So the runtime will indeed generate a name for this consumer group / subscription if it isn't supplied

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-pulsar.md [hunk 1]
```diff
@@ -66,6 +68,7 @@ spec:
 | enableTLS          | N  | Enable TLS.  Default: `"false"` | `"true"`, `"false"` |
 | token              | N  | Enable Authentication.  | [How to create pulsar token](https://pulsar.apache.org/docs/en/security-jwt/#generate-tokens)|
 | tenant             | N  | The topic tenant within the instance. Tenants are essential to multi-tenancy in Pulsar, and spread across clusters.  Default: `"public"` | `"public"` |
+| consumerID         | N  | Used for subscriptions.  | `"certification"`
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-pulsar.md]:
```suggestion
| consumerID         | N  | Used to set the subscription name or consumer ID.  | `"topic1"`
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-pulsar.md [hunk 2]
```diff
@@ -28,6 +28,8 @@ spec:
     value: "public"
   - name: token
     value: "eyJrZXlJZCI6InB1bHNhci1wajU0cXd3ZHB6NGIiLCJhbGciOiJIUzI1NiJ9.eyJzd"
+  - name: consumerID
+    value: "certification"
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-pulsar.md]:
```suggestion
    value: "topic1"
```

**Review by msfussell — APPROVED** [review]

**Pull request merged**